roles/openshift_health_checker/openshift_checks/logging/podlogs.py

```diff
@@ -0,0 +1,143 @@
+
+        es_pods = super(PodLogQueryTime, self).running_pods(pods)
+        if not len(es_pods):
+            msg = ('No Elasticsearch pods were found to be in the "Running" state.'
+                   'At least one Elasticsearch pod is required in order to perform this check.')
+            return {"failed": True, "changed": False, "msg": msg}
+
+        failed = False
+        msg = ""
+
+        try:
+            uid = self.curl_kibana_with_uuid(kibanna_pods[0], task_vars)
+            self.wait_until_cmd_or_err(es_pods[0], uid, pod_log_timeout, task_vars)
+        except OpenShiftCheckException as error:
+            failed = True
+            msg = str(error)
+        except KeyError as error:
+            failed = True
+            msg = "Invalid json response from pod: {}".format(str(error))
+        except ValueError as error:
+            failed = True
+            msg = "Invalid response from pod {}".format(str(error))
+
+        return {"failed": failed, "msg": msg}
+
+    def wait_until_cmd_or_err(self, es_pod, uid, timeout_secs, task_vars):
+        """Wait a maximum of timeout_secs for the uuid logged in Kibana to be found in
+        the Elasticsearch logs. Since we are querying for a message with the the unique
+        uuid from earlier, there should only be a single match.
+        Raise an OpenShiftCheckException if the timeout is reached before a result is found."""
+
+        interval = 1  # amount of time to wait between retries
+        orig_timeout = timeout_secs
+
+        while timeout_secs > 0:
+            time.sleep(interval)
+            timeout_secs -= interval
+
+            total = self.query_es_from_es(es_pod, uid, task_vars)
+            if total == 1:
+                return
+
+        msg = "expecting match in Elasticsearch for message with uuid {}, but no matches were found after {}s."
+        raise OpenShiftCheckException(msg.format(uid, orig_timeout))
+
+    def curl_kibana_with_uuid(self, kibanna_pod, task_vars):
+        """curl Kibana with a unique uuid."""
+        uid = str(uuid4())
+        pod_name = kibanna_pod["metadata"]["name"]
+        exec_cmd = "exec {pod_name} -c kibana -- curl --connect-timeout 1 -s http://localhost:5601/{uid}"
+
+        error_str = super(PodLogQueryTime, self).exec_oc(
+            self.execute_module,
+            self.logging_namespace,
+            exec_cmd.format(pod_name=pod_name, uid=uid),
+            [], task_vars,
+        )
+        error_code = json.loads(error_str).get("statusCode")
+
+        if error_code != 404:
+            msg = "expecting server error code 404, but got {} instead."
+            raise OpenShiftCheckException(msg.format(error_code))
+
+        return uid
+
+    def query_es_from_es(self, es_pod, uid, task_vars):
+        """curl the Elasticsearch pod and look for a unique uuid in its logs."""
+        pod_name = es_pod["metadata"]["name"]
+        exec_cmd = ("exec {pod_name} -- curl --connect-timeout 1 -s -k -f "
```

Comment:
Thanks @richm. I see no reason not to use `--cacert /etc/elasticsearch/secret/admin-ca` then.

As for the `--connect-timeout`, 1 second seems too aggressive.
If we're willing to wait up to 10 minutes retrying until this succeeds, I'd suggest being more tolerant with slow connections.
`--connect-timeout 60`?